klient/wait/conditions/conditions.go

```diff
@@ -303,3 +303,17 @@ func (c *Condition) DeploymentAvailable(name, namespace string) apimachinerywait
 		v1.ConditionTrue,
 	)
 }
+
+// DaemonSetReady is a helper function used to check if a daemonset's pods are scheduled and ready
+func (c *Condition) DaemonSetReady(daemonset k8s.Object) apimachinerywait.ConditionWithContextFunc {
```

Comment:
@cartermckinnon ... based on the description in comment, should name be `DeamonSetPodReady` ? Are there other conditions that can be checked for a ready status for `DaemonSet` ?

Reply:
The `DaemonSet.Status` *does* have a `Condition` in the API types, but AFAICT, it's unimplemented, the DaemonSet controller doesn't populate it: https://github.com/search?q=repo%3Akubernetes%2Fkubernetes%20DaemonSetConditionType&type=code

I don't feel too strongly about the name, SGTM if you think that's clearer.

Reply:
@cartermckinnon when I looked, i came with same conclusion that conditions not populated... so let's keep name you have. Plus, as long as it's documented, SGTM.